airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/DataTypeTestArgumentProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.standardtest.destination;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataTypeTestArgumentProvider implements ArgumentsProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataTypeTestArgumentProvider.class);
+
+  public static final CatalogMessageTestConfigWithCompatibility BASIC_TEST =
+      new CatalogMessageTestConfigWithCompatibility("data_type_basic_test_catalog.json", "data_type_basic_test_messages.txt",
+          new TestCompatibility(true, false, false));
+  public static final CatalogMessageTestConfigWithCompatibility ARRAY_TEST =
+      new CatalogMessageTestConfigWithCompatibility("data_type_array_test_catalog.json", "data_type_array_test_messages.txt",
+          new TestCompatibility(true, true, false));
+  public static final CatalogMessageTestConfigWithCompatibility OBJECT_TEST =
+      new CatalogMessageTestConfigWithCompatibility("data_type_object_test_catalog.json", "data_type_object_test_messages.txt",
+          new TestCompatibility(true, false, true));
+  public static final CatalogMessageTestConfigWithCompatibility OBJECT_WITH_ARRAY_TEST =
+      new CatalogMessageTestConfigWithCompatibility("data_type_array_object_test_catalog.json", "data_type_array_object_test_messages.txt",
+          new TestCompatibility(true, true, true));
+
+  @Override
+  public Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception {
+    return Stream.of(
+        getArguments(BASIC_TEST),
+        getArguments(ARRAY_TEST),
+        getArguments(OBJECT_TEST),
+        getArguments(OBJECT_WITH_ARRAY_TEST));
+  }
+
+  private Arguments getArguments(CatalogMessageTestConfigWithCompatibility testConfig) {
+    return Arguments.of(testConfig.messageFile, testConfig.catalogFile, testConfig.testCompatibility);
+  }
+
+  public record TestCompatibility(boolean requireBasicCompatibility,
```

Comment:
checking my understanding: once the items in [your comment](https://github.com/airbytehq/airbyte/pull/10325#issuecomment-1046886134) are resolved, this will be true for all destinations?

Reply:
Right, we need to upgrade our abstract test part for the proper result validation. As it's a common part, we apply it for all future destinations after we do for one destination.